tools/lab_dispatch/generic_source.yaml

```diff
@@ -0,0 +1,78 @@
+job_queue: $INPUT_QUEUE
+global_timeout: 3600
+output_timeout: 1800
+provision_data:
+  $INPUT_DATA_SOURCE
+test_data:
+  attachments:
+    - local: "tools/lab_dispatch/resources/manifest.conf"
+      agent: "manifest.conf"
+    - local: "tools/lab_dispatch/resources/input.conf"
+      agent: "input.conf"
+    - local: "tools/lab_dispatch/resources/service_file.service"
+      agent: "service_file.service"
+    - local: "tools/lab_dispatch/build_install_deb.py"
+      agent: "build_install_deb.py"
+  test_cmds: |
+    #!/usr/bin/env bash
+
+    set -x
+
+    # input arguments
+    CHECKBOX_REVISION=$INPUT_CHECKBOX_REVISION
+    RESOURCES_PATH="attachments/test"
+
+    TOOLS_PATH=tools
+
+    # retrieve all scripts/tools necessary from a repo
+    curl -Ls -o install_tools.sh https://raw.githubusercontent.com/canonical/hwcert-jenkins-tools/main/install_tools.sh
+    source install_tools.sh $TOOLS_PATH
+
+    # ensure device is available before continuing
+    wait_for_ssh
+
+    _run sudo add-apt-repository ppa:checkbox-dev/edge
+    _run install_packages git python3 python3-pip dpkg-dev
+
+    wait_for_ssh
+
+    _put $RESOURCES_PATH/build_install_deb.py :
+    _run git clone https://github.com/canonical/checkbox.git
+    _run git -C checkbox checkout $CHECKBOX_REVISION
+    _run python3 build_install_deb.py --clean checkbox/checkbox-ng \
+        checkbox/checkbox-support checkbox/providers/resource \
+        checkbox/providers/base checkbox/providers/sru
+    _run sudo systemctl restart checkbox-ng.service
+
+    git clone https://github.com/canonical/checkbox.git
+    git -C checkbox checkout $CHECKBOX_REVISION
+
+    python3 -m venv --system-site-packages checkbox_venv
+    . checkbox_venv/bin/activate
+    pip install --upgrade pip setuptools
+    pip install checkbox/checkbox-ng
+    # retrieve manifest
+    MANIFEST_FILE=manifest.conf
+    fetch_manifest --manifest_file manifest.conf $CID $HEXR_DEVICE_SECURE_ID
+    if [ $? -ne 0 ]; then
+        echo "Using default manifest"
```

Comment:
Ha, this is funny. You might be interested in [this PR](https://github.com/canonical/hwcert-jenkins-jobs/pull/583) (currently in draft). I think that, given the current status of things, it would be a good idea to deprecate the use of a default manifest.

Reply:
for this PR I think it makes sense to have a default or if we even need it, a way to actually provide a manifest via a parameter because this is a testing workflow, so using it for new queues, or to test, for example, new manifest keys, is very much on the table. This could be a very interesting follow up direction actually.

Reply:
as for your pr, I agree that in sru it should fail and blow up if the manifest doesn't exist, else you will spend time chasing ghosts sometimes (or wondering why the test coverage is so low)